### ISSUE TITLE
`aliasing.forbid` Traceback Error

### ISSUE THREAD

**Issue by tac14yo** [sqlfluff/sqlfluff#5954]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When running a sqlfluff lint on a file the return was a Warning Error

`WARNING    Unable to lint models\test_file.sql due to an internal error. Please report this as an issue with your query's contents and stacktrace below!
To hide this warning, add the failing file to .sqlfluffignore
Traceback (most recent call last):
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\linter\runner.py", line 110, in run
    yield partial()
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\linter\linter.py", line 700, in lint_rendered
    return cls.lint_parsed(
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\linter\linter.py", line 674, in lint_parsed
    formatter.dispatch_file_violations(
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\cli\formatters.py", line 262, in dispatch_file_violations
    s = self._format_file_violations(
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\cli\formatters.py", line 241, in _format_file_violations
    self.format_violation(
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\cli\formatters.py", line 426, in format_violation
    v_dict = violation.to_dict()
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\errors.py", line 304, in to_dict
    fixes=[fix.to_dict() for fix in self.fixes],
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\errors.py", line 304, in <listcomp>
    fixes=[fix.to_dict() for fix in self.fixes],
  File "C:\Users\tschmidt\Documents\Data_Files\VS_Code\dbt-env\dbt-180-cli\lib\site-packages\sqlfluff\core\rules\fix.py", line 149, in to_dict
    assert len(self.edit[0].source_fixes) == 1
AssertionError`

### Expected Behaviour

The file should have had linting violations found and documented instead the script failed.

### Observed Behaviour

When running a similar model (lets call it test_file):

```
with
foo as (
    select * from vee
),

bar as (
    select * from baz
),

final as (
    select
        foo.col1,
        foo.col2,
        foo.col3,
        bar.col4,
        bar.col5,
        bar.col6,
        bar.col7
    from
        foo as foo
    left join
        bar
        on
            foo.col1 = bar.col2
)

select * from final
```
After running `sqlfluff lint models\test_file.sql` I get the earlier mentioned error. 

When I alter the code to remove the line `foo as foo` to either be `foo as t1` or just `foo` and run the above lint again I get no errors.

Therefore, it seems the issue is directly related to the table being aliased as itself. I did also run a test on an older version of sqlfluff `2.3.2` and there is no issue with it being aliased as itself and it just removes it as it should.

This makes me wonder if it has something to do with code introduced for `aliasing.self_alias.column`, however running `sqlfluff lint models\test_file.sql --exclude-rules AL09` still throws the error.

### How to reproduce

Use the above sql code and run `sqlfluff lint` and use the below configs

[sqlfluff]
dialect = redshift

[sqlfluff:rules:aliasing.forbid]
force_enable = True

### Dialect

Redshift

### Version

sqlfluff, version 3.0.7
Python 3.9.7

### Configuration

[sqlfluff]
dialect = redshift

[sqlfluff:rules:aliasing.forbid]
force_enable = True

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
